cmd/mt-replicator/consume.go

```diff
@@ -43,30 +43,44 @@ func NewConsumer(brokers []string, group, topic string) (*Consumer, error) {
 
 func (c *Consumer) Consume(publisher *Publisher) {
 	buf := make([]*schema.MetricData, 0)
-	for m := range c.consumer.Messages() {
-		md := &schema.MetricData{}
-		_, err := md.UnmarshalMsg(m.Value)
-		if err != nil {
-			log.Error(3, "kafka-mdm decode error, skipping message. %s", err)
-			continue
-		}
-		buf = append(buf, md)
-		if len(buf) > 1000 {
-			complete := false
-			for !complete {
-				if err = publisher.Send(buf); err != nil {
-					log.Error(3, "failed to publish %d metrics. trying again in 1second", len(buf))
-					time.Sleep(time.Second)
-				} else {
-					complete = true
+	ticker := time.NewTicker(time.Second * 10)
+	counter := 0
+	msgChan := c.consumer.Messages()
+	defer close(c.Done)
+	for {
+		select {
+		case m, ok := <-msgChan:
+			if !ok {
+				return
+			}
+			md := &schema.MetricData{}
+			_, err := md.UnmarshalMsg(m.Value)
+			if err != nil {
+				log.Error(3, "kafka-mdm decode error, skipping message. %s", err)
+				continue
+			}
+			counter++
+			buf = append(buf, md)
+			if len(buf) > 1000 {
+				log.Debug("flushing metricData buffer to kafka.")
+				complete := false
+				for !complete {
+					if err = publisher.Send(buf); err != nil {
+						log.Error(3, "failed to publish %d metrics. trying again in 1second", len(buf))
+						time.Sleep(time.Second)
+					} else {
+						complete = true
+					}
 				}
+				buf = buf[:0]
+				c.consumer.MarkPartitionOffset(m.Topic, m.Partition, m.Offset, "")
 			}
-			buf = buf[:0]
-			c.consumer.MarkPartitionOffset(m.Topic, m.Partition, m.Offset, "")
+		case <-ticker.C:
+			log.Info("%d metrics procesed in last 10seconds.", counter)
```

Comment:
I think it's very possible that the prior case will keep the select case rather busy (especially when it hits a failure and sleeps a bit), in which case this ticker can silently drop ticks. This is not a big issue in this case, except we shouldn't claim it's `in last 10seconds`. it might very well be in the last 20s or 30s. so i would just remove that part, we can just use the timestamps of the messages for guidance. (or actually measure how long it took since last time)